nntrainer/src/neuralnet.cpp

```diff
@@ -890,7 +890,12 @@ int NeuralNetwork::addLayer(std::shared_ptr<Layer> layer) {
   LayerType type = layer->getType();
   if (type == LAYER_UNKNOWN)
     return ML_ERROR_INVALID_PARAMETER;
-  layers.push_back(layer);
+
+  if (layers.size() > 1 && layers[layers.size() - 1]->getType() == LAYER_LOSS) {
```

Comment:
For readability, How about just adding extra member variable or method, instead of this expression?

I think you can set some variable like `compiled = true` in `NeuralNetwork::init()`

Reply:
that sounds better, updated as per your suggestion with added variable `initialized`.